CSC322-Grinnell/mayflower-directory
# Staffs' MVC and Departments Views, Combined

This has gotten big enough that it warrants a PR on its own.

# Changes

- Implements MVC for staff
- Implements `has many` relationship between departments and staff
- Implements dynamic *Staff* section under each department
- Utilizes "cardview" from #217 to containerize each "gallery", i.e. department
- Externalizes profile (from profiles/index.html.erb) as standalone layout, i.e. as a partial called layouts/_profile_cardview.html.erb

# Testing

1. `git checkout departments-staffs-views`
2. `rails db:migrate`
2. `rails s -b $IP -p $PORT`
3. go under staff
4. create some staff profiles 
4. make sure the association with departments show up under the newly created profile
4. make sure the staff show up under their associated department page
4. make sure departments/index, ./show, ./edit, ./new looks fine
4. make sure staffs/index, ./show, ./edit, ./new looks fine

# Known Issues

## departments/n/edit 
- ~*Remove Association* button under staff section doesn't work. That has not been implemented yet~
- *Add New* redirects to *staffs* because no existing view or logic support add new
- A tiny red button shows up under the *Add New* button
- Image upload may or may not work

## Miscellaneous
- Image upload needs to be re-enabled for departments
- Image upload needs to be implemented for staffs
- Description and contact fields under department strips line breaks and do not display them after form save.
